lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,227 @@
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const drawer = getDrawer()!;
+
+    expect(drawerWithAddedClassName).toHaveClass('foo');
+
+    for (const cls of Array.from(drawer.classList)) {
+      expect(drawerWithAddedClassName).toHaveClass(cls);
+    }
+  });
+
+  it('includes any passed in attributes to the dialog element', function() {
+    const drawer = getDrawer({ id: 'drawer-id', lang: 'en_US' });
+    expect(drawer).toHaveAttribute('id', 'drawer-id');
+    expect(drawer).toHaveAttribute('lang', 'en_US');
+  });
+
+  describe('NxDrawer event listener support', () => {
+    it('executes onClose callback when ESC key is pressed', async function() {
+      const user = userEvent.setup();
+      const mockOnClose = jest.fn();
+
+      quickRender({ onClose: mockOnClose });
+
+      expect(mockOnClose).not.toHaveBeenCalled();
+
+      await user.keyboard('{Escape}');
+
+      expect(mockOnClose).toHaveBeenCalled();
+    });
+
+    it('does not execute onClose if a key other than ESC is pressed', async function () {
+      const user = userEvent.setup();
+
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const drawer = getDrawer({ onClose: mockOnClose })!;
+
+      await user.keyboard('Tab');
+      await user.keyboard('Enter');
+      await user.keyboard('q');
+      await user.keyboard('Q');
+
+      fireEvent.animationEnd(drawer);
+
+      expect(mockOnClose).not.toHaveBeenCalled();
+    });
+
+    it('executes onCancel when drawer is closed and animation is completed', async function() {
+      const user = userEvent.setup();
+      const mockOnCancel = jest.fn();
+
+      const Fixture = () => {
+        const [open, toggleDrawer] = useToggle(true);
+        return (
+          <NxDrawer open={open} onClose={toggleDrawer} onCancel={mockOnCancel}>
+            <NxDrawer.Header>
+              <NxDrawer.HeaderTitle>Title</NxDrawer.HeaderTitle>
+            </NxDrawer.Header>
+          </NxDrawer>
+        );
+      };
+
+      render(<Fixture />);
+
+      await act(async () => {
+        const closeButton = screen.getByRole('button', { name: 'Close', hidden: true });
+        await user.click(closeButton);
+      });
+
+      const dialog = screen.getByRole('dialog', { hidden: true });
+      await fireEvent.animationEnd(dialog);
+
+      expect(mockOnCancel).toHaveBeenCalled();
+    });
+
+    it('executes onClose when clicked outside of the drawer', async function() {
+      const user = userEvent.setup();
+      const mockOnClose = jest.fn();
+
+      render(
+        <div>
+          <div>Outside</div>
+          <NxDrawer open={true} onClose={mockOnClose}>
+            <NxButton>Inside</NxButton>
+          </NxDrawer>
+        </div>
+      );
+
+      const outsideDiv = screen.getByText('Outside');
+      const insideButton = screen.getByRole('button', { name: 'Inside', hidden: true });
```

Comment:
I've been investigating why all of the drawer children are being considered as hidden and I realized it's because JSDOM doesn't have full native support for dialog, and so follows the code path where `show()` does not get called.  It does however have support for toggling the `open` property manually, and apparently RTL considers that property when deciding whether things are hidden.  In 0a7024b41, I add support for manually toggling the `open` property in browsers that don't support `show()` and `close()`.  This enabled me to get rid of all these `hidden: true` checks.

However, while working on this I did also realize that we don't have unit tests focusing on the `open` prop itself, and we should.  For instance, a test that if you start with `open={true}` then the `open` property on the DOM element will immediately be `true`, but that if you start with it `false` and then change it to true the DOM element property will only
change after animation completes.  And then similarly for starting with `false` vs transitioning from `true` to `false`.